libraries/AP_HAL_SITL/SITL_cmdline.cpp

```diff
@@ -287,7 +288,7 @@ void SITL_State::_parse_command_line(int argc, char * const argv[])
 
     GetOptLong gopt(argc, argv, "hwus:r:CI:P:SO:M:F:c:",
                     options);
-    while ((opt = gopt.getoption()) != -1) {
+    while (!is_replay && (opt = gopt.getoption()) != -1) {
```

Comment:
Maybe just nitpicking, but can't we just move the if (is_replay) here and put the while loop into the the else cause ?
That would also have the advantage to allow us to define another parser for the SITL replay configuration